File: linked-list/return-k-element/solution_v1_test.go
package return_k_element

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestReturnKElement_1(t *testing.T) {
	nodeFour := Node{
		Next:  nil,
		value: 4,
	}
	nodeThree := Node{
		Next:  &nodeFour,
		value: 3,
	}
	nodeTwo := Node{
		Next:  &nodeThree,
		value: 2,
	}
	nodeOne := Node{
		Next:  &nodeTwo,
		value: 1,
	}
	assert.Equal(t, ReturnKElement(&nodeOne), 4)
}

func TestReturnKElement_2(t *testing.T) {
	nodeOne := Node{
		Next:  nil,
		value: 1,
	}
	assert.Equal(t, ReturnKElement(&nodeOne), 1)
}

File: sum-list-java/sum-list-java/src/test/java/SolutionV1Test.java
import org.junit.Assert;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

public class SolutionV1Test {


    @Test
    public void sumListTest() {
        Node nodeSix = new Node(null, 2);
        Node nodeFive = new Node(nodeSix, 9);
        Node nodeFour = new Node(nodeFive, 5);

        Node nodeThree = new Node(null, 6);
        Node nodeTwo = new Node(nodeThree, 1);
        Node nodeOne = new Node(nodeTwo, 7);

        SolutionV1 solutionV1 = new SolutionV1();
        Node response = solutionV1.sumList(nodeOne, nodeFour);

        List<Integer> expected = new ArrayList<>();
        extracted(response, expected);
        Assert.assertEquals(expected, List.of(new Integer[]{9, 1, 2}));
    }

    private static void extracted(Node node, List<Integer> expected) {
        expected.add(node.data);

        if (node.next == null) {
            return;
        }
        extracted(node.next, expected);
    }
}
